src/storage-types/src/dyncfgs.rs

```diff
@@ -32,9 +32,25 @@ pub const STORAGE_DOWNGRADE_SINCE_DURING_FINALIZATION: Config<bool> = Config::ne
     during shard finalization",
 );
 
+/// Rules for enriching the `client.id` property of Kafka clients with
+/// additional data.
+///
+/// The configuration value must be a JSON array of objects containing keys
+/// named `pattern` and `payload`, both of type string. Rules are checked in the
+/// order they are defined. If the rule's pattern matches the address of any
+/// broker in the connection, then the payload is appended to the client ID. A
+/// rule's payload is always prefixed with `-`, to separate it from the
+/// preceding data in the client ID.
```

Comment:
I'd be nice to explicitly say that a pattern is a regular expression (instead of say just wildcards) and have an example

Reply:
Will do.

Reply:
Done.